waves-integration/src/main/scala/com/wavesplatform/dex/tool/KamonTraceUtils.scala

```diff
@@ -0,0 +1,111 @@
+package com.wavesplatform.dex.tool
+
+import akka.http.scaladsl.server.Directives.mapInnerRoute
+import akka.http.scaladsl.server.Route
+import kamon.Kamon
+import kamon.context.{BinaryPropagation, Context, Storage}
+import kamon.trace.Span
+import kamon.trace.Trace.SamplingDecision
+import kamon.trace.Tracer.PreFinishHook
+
+import java.io.ByteArrayOutputStream
+import scala.concurrent.{ExecutionContext, Future}
+
+object KamonTraceUtils {
+
+  def setSpanNameAndForceSamplingDecision(name: String): Unit = {
+
+    //initially span has Unknown state and there is no 100% guarantee that it will be logged in jaeger
+    //so we have to force sampling decision (apply our sampler to a span) to force its state
+    //Unknown -> Sample
+    //Unknown -> DoNotSample
+    //in case when a span is marked as Sample it will be sent in jaeger.
+    Kamon.currentSpan().takeSamplingDecision()
+
+    Kamon.currentSpan().name(name)
+  }
+
+  def mkTracedRoute(operationName: String)(route: Route): Route = mapInnerRoute { route => ctx =>
+    setSpanNameAndForceSamplingDecision(operationName)
+    route(ctx)
+  }(route)
+
+  //https://github.com/kamon-io/Kamon/issues/829
+  def propagateTraceCtxThroughCachedFuture[A](future: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    val span = mkIgnoredSpan()
+    val scope = Kamon.storeContext(Kamon.currentContext().withEntry(Span.Key, span))
+
+    try future.transform(
+      res => {
+        finishSpanAndContextScope(span, scope)
+        res
+      },
+      err => {
+        failSpanAndContextScope(span, scope, err)
+        err
+      }
+    )
+    catch {
+      case e: Throwable =>
+        failSpanAndContextScope(span, scope, e)
+        throw e
+    }
+  }
+
+  def runWithIgnoredSpan[A](f: => A): A = {
+    val span = mkIgnoredSpan()
+    Kamon.runWithSpan(span, finishSpan = true)(f)
```

Comment:
It looks like the `runWithSpan` with one parameter.

Reply:
fixed